libvast/test/detail/column_iterator.cpp

```diff
@@ -0,0 +1,81 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#define SUITE column_iterator
+
+#include "vast/detail/column_iterator.hpp"
+
+#include "vast/test/test.hpp"
+
+#include <numeric>
+
+#include "vast/detail/range.hpp"
+
+using namespace vast;
+
+namespace {
+
+using column_range = detail::iterator_range<detail::column_iterator<int>>;
+
+struct fixture {
+  std::vector<int> buf;
+
+  fixture() {
+    buf.resize(16);
+    std::iota(buf.begin(), buf.end(), 0);
+  }
+
+  column_range column(size_t rows, size_t col) {
+    detail::column_iterator<int> first{buf.data() + col, rows};
+    return {first, first + (16 / rows)};
+  }
+};
+
+} // namespace <anonymous>
+
+#define CHECK_RANGE(rng, ...)                                                  \
+  {                                                                            \
+    std::vector<int> xs{rng.begin(), rng.end()};                               \
+    std::vector<int> ys{__VA_ARGS__};                                          \
+    CHECK_EQUAL(xs, ys);                                                       \
+  }
+
+FIXTURE_SCOPE(column_iterator_tests, fixture)
+
+TEST(four by four) {
+  MESSAGE("visit buf as if 4x4 matrix");
```

Comment:
Should this (and the messages below) read `visit buf as 4x4 matrix`, without the `if`?